src/components/CategoryList/CategoryListMobile.js

```diff
@@ -0,0 +1,52 @@
+import { useState } from 'react';
+
+import { useTheme } from '@mui/material/styles';
+import { useMediaQuery } from '@mui/material';
+import Grid from '@mui/material/Unstable_Grid2';
+
+import CategoryList from './CategoryList';
+import CategoryMoreButton from '../CategoryMoreButton/CategoryMoreButton'
+
+const CategoryListMobile = (props) => {
+  const { array, onClick } = props;
+  const theme = useTheme();
+  const sm = useMediaQuery(theme.breakpoints.up('sm'));
+  const md = useMediaQuery(theme.breakpoints.up('md'));
+  const [showMore, setShowMore] = useState(false);
+  const [buttonText, setButtonText] = useState('Більше категорій');
```

Comment:
Don't create unnecessary useState. Show specific text according to the showMore variable.